src/serialization/sexp.lisp

```diff
@@ -249,12 +245,79 @@ s-expressions."))
+
+(defgeneric get-slot-definition (class slot-name) ; From moptilities
+  (:documentation "Returns the slot-definition for the slot named `slot-name` in the class specified by `class-specifier`. Also returns \(as a second value\) true if the slot is an indirect slot of the class.")
+  (:method ((class-specifier t) slot-name)
+    (let ((class (get-class class-specifier)))
+      (let* ((indirect-slot? nil)
+             (slot-info
+               (or (find slot-name (class-direct-slots class)
+                         :key #'slot-definition-name)
+                   (and (setf indirect-slot? t)
+                        (find slot-name (class-slots class)
+                              :key #'slot-definition-name)))))
+        (values slot-info indirect-slot?)))))
+
+(defgeneric slot-properties (class-specifier slot-name) ; From moptilities
+  (:documentation "Returns a property list describing the slot named slot-name in class-specifier.")
+  (:method ((class symbol) slot-name)
+           (slot-properties (find-class class) slot-name))
+  (:method ((object standard-object) slot-name)
+           (slot-properties (class-of object) slot-name))
+  (:method ((class class) slot-name)
+           (declare (ignorable slot-name))
+           (multiple-value-bind (slot-info indirect-slot?)
+                                (get-slot-definition class slot-name)
+             `(:name ,(slot-definition-name slot-info)
+                     ,@(when (eq (slot-definition-allocation slot-info) :class)
+                         `(:allocation ,(slot-definition-allocation slot-info)))
+                     :initargs ,(slot-definition-initargs slot-info)
+                     :initform ,(slot-definition-initform slot-info)
+                     ,@(when (and (not (eq (slot-definition-type slot-info) t))
+                                  (not (eq (slot-definition-type slot-info) nil)))
+                         `(:type ,(slot-definition-type slot-info)))
+                     ,@(unless indirect-slot?
+                         `(:readers ,(slot-definition-readers slot-info)
+                                    :writers ,(slot-definition-writers slot-info)))
+                     :documentation ,(documentation slot-info t)))))
+
 (defgeneric deserialize-class (class-symbol slots deserialized-objects)
-  (:documentation "Read and return an the instance corresponding to CLASS-SYMBOL with SLOTS."))
+  (:documentation "Read and return an the instance corresponding to CLASS-SYMBOL with SLOTS.
+SLOTS is a list of pairs: the first element is the slot name (a symbol) and the
+second element its value."))
 
 (defmethod deserialize-class ((class-symbol t) slots deserialized-objects)
-  (let ((object (make-instance class-symbol)))
-    object))
+  (let* ((no-initarg-slots '())
+         (instance (apply 'make-instance
+                          class-symbol
+                          (apply
+                           'append
+                           (delete nil
```

Comment:
No need to delete nil, because append will skip them:

```
CL-USER> (append (list 1 2)
                 nil
                 (list 3 4))
(1 2 3 4)
```

Or do I miss something?

Reply:
Oh right, left-over of some other code.  Can you remove it?